typing/typecore.ml

```diff
@@ -4516,69 +4514,6 @@ and type_expect_
       | _ ->
           assert false
       end
-  | Pexp_letmodule(name, smodl, sbody) ->
-      let lv = get_current_level () in
-      let (id, pres, modl, _, body) =
-        with_local_level_generalize begin fun () ->
-          let modl, pres, id, new_env =
-            Typetexp.TyVarEnv.with_local_scope begin fun () ->
-              let modl, md_shape = !type_module env smodl in
-              Mtype.lower_nongen lv modl.mod_type;
-              let pres =
-                match modl.mod_type with
-                | Mty_alias _ -> Mp_absent
-                | _ -> Mp_present
-              in
-              let scope = create_scope () in
-              let md_uid = Uid.mk ~current_unit:(Env.get_current_unit ()) in
-              let md_shape = Shape.set_uid_if_none md_shape md_uid in
-              let md =
-                { md_type = modl.mod_type; md_attributes = [];
-                  md_loc = name.loc;
-                  md_uid; }
-              in
-              let (id, new_env) =
-                match name.txt with
-                | None -> None, env
-                | Some name ->
-                    let id, env =
-                      Env.enter_module_declaration
-                        ~scope ~shape:md_shape name pres md env
-                    in
-                    Some id, env
-              in
-              modl, pres, id, new_env
-            end
-          in
-          (* Ideally, we should catch Expr_type_clash errors
-             in type_expect triggered by escaping identifiers
-             from the local module and refine them into
-             Scoping_let_module errors
-           *)
-          let body = type_expect new_env sbody ty_expected_explained in
-          (id, pres, modl, new_env, body)
-        end
-        ~before_generalize: begin fun (_id, _pres, _modl, new_env, body) ->
-          (* Ensure that local definitions do not leak. *)
-          (* required for implicit unpack *)
-          enforce_current_level new_env body.exp_type
```

Comment:
So, is it safe to remove this ? (I guess so since the [related test](https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/typing-implicit_unpack/implicit_unpack.ml) passes)